tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_NormalVectorQuantities.cpp

```diff
@@ -0,0 +1,298 @@
+      std::move(normal_covector_quantities));
+}
+
+template <bool UseFlatSpace, size_t Dim, typename DbTagsList>
+void check_normal_covector_quantities(
+    const gsl::not_null<db::DataBox<DbTagsList>*> box) noexcept {
+  using internal_directions = domain::Tags::InternalDirections<Dim>;
+  const auto& unnormalized_normal_covectors = db::get<domain::Tags::Interface<
+      internal_directions, domain::Tags::UnnormalizedFaceNormal<Dim>>>(*box);
+  for (const auto& [direction, neighbors] :
+       db::get<domain::Tags::Element<Dim>>(*box).neighbors()) {
+    (void)neighbors;
+    Variables<tmpl::conditional_t<
+        UseFlatSpace,
+        tmpl::list<
+            evolution::dg::Actions::detail::OneOverNormalVectorMagnitude>,
+        tmpl::list<evolution::dg::Actions::detail::OneOverNormalVectorMagnitude,
+                   evolution::dg::Actions::detail::NormalVector<Dim>,
+                   InverseSpatialMetric<Dim>>>>
+        fields_on_face{
+            db::get<domain::Tags::Interface<internal_directions,
+                                            domain::Tags::Mesh<Dim - 1>>>(*box)
+                .at(direction)
+                .number_of_grid_points()};
+    if constexpr (not UseFlatSpace) {
+      double temp = 0.1;
+      for (auto& component : get<InverseSpatialMetric<Dim>>(fields_on_face)) {
+        for (double& t : component) {
+          t = temp;
+          temp *= 1.1;
+        }
+      }
+    }
+
+    evolution::dg::Actions::detail::normal_covector_quantities<
+        tmpl::conditional_t<UseFlatSpace, FlatSpaceSystem,
+                            SystemWithInverseMetric<Dim>>>(
+        box, make_not_null(&fields_on_face), direction,
+        unnormalized_normal_covectors);
+    if constexpr (UseFlatSpace) {
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          get(magnitude(
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  *get<evolution::dg::Tags::InternalFace::
+                           NormalCovectorQuantities<Dim>>(*box)
+                       .at(direction)))));
+    } else {
+      const auto& normal_covector =
+          get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+              *get<evolution::dg::Tags::InternalFace::NormalCovectorQuantities<
+                   Dim>>(*box)
+                   .at(direction));
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          sqrt(get(
+              dot_product(normal_covector, normal_covector,
+                          get<InverseSpatialMetric<Dim>>(fields_on_face)))));
+    }
+
+    for (const double t :
+         get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+             *get<evolution::dg::Tags::InternalFace::NormalCovectorQuantities<
+                  Dim>>(*box)
+                  .at(direction)))) {
+      CHECK(t > 0.0);
```

Comment:
Just so I understand, why is t not allowed to be zero or negative?

Reply:
`t` is the magnitude of the normal covector, so it I think needs to be non-negative

Reply:
In that case, how about "covector_magnitude" or something? It wasn't obvious to me from the name what t actually was